### DISCUSSION TITLE
Add pathName Prop In Server Side Page Component

### DISCUSSION THREAD

**Discussion by waseem-sajjad** [vercel/next.js#66653] [Ideas]
### Goals

1. Add request pathName is server side page component to avoid adding Middleware 

### Background

to avoid adding Middleware

### Proposal

i have implemented yet.

**Comment by icyJoseph** [comment]
You can do it like this, https://github.com/vercel/next.js/discussions/63298#discussioncomment-8797531

**Comment by devajakamerus** [comment]
import { staticGenerationAsyncStorage } from "next/dist/client/components/static-generation-async-storage.external";
 in component and access url ( or any other request property) from staticGenerationAsyncStorage.getStore()

Surely they could offer cleaner api for this since it's a feature people need sometimes.
